### PR TITLE
started adding insilico stuff

### PR DESCRIPTION
**Pull request by alvaralmstedt**
Added: Sex & version tag to qc report. insilico reports created for all three options (tn no to). Normal is used for insilico coverage reports when availiable.

TODO: Add the insilico reports to the main qc excel

### CODE REVIEW

File: configs/cluster.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 __default__:
-    queue: wgs.q
+    queue: test.q
```
Comment (hannasoderstrom1) [inline configs/cluster.yaml:2]: I guess you have changed to test.q for testing and will change back later?

File: workflows/rules/qc/insilico_coverage.smk [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+def get_bedfile_path(wcs):
+    return config["insilico"][f"{wcs.insiliconame}"]['bedfile']
+
+def get_bedfile_version(wcs):
+    return config["insilico"][f"{wcs.insiliconame}"]['version']
+
+rule insilico_coverage:
+    input:
+        bamfile = "{workingdir}/dedup/{sampleid}_DEDUP.bam"
```
Comment (hannasoderstrom1) [inline workflows/rules/qc/insilico_coverage.smk:14]: the files are under {workingdir}/{stype}/dedup/{sname}_DEDUP.bam
(stype = tumor or normal)

File: workflows/scripts/insilico_coverage/anybelow_sample.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        reader = csv.DictReader(inp, delimiter=",", fieldnames=fieldnames)
+        for i in reader:
+            if int(i["cov"]) >= limit:
+                continue
+            else:
+                below_limit.append(i)
+    return below_limit
+
+
+# Takes the csv object containing only the regions below and merges adjacent positions into a single region
```
Comment (hannasoderstrom1) [inline workflows/scripts/insilico_coverage/anybelow_sample.py:22]: regions below what?

File: workflows/scripts/insilico_coverage/anybelow_sample.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                writer.writerow({"chr": str(i["chr"]),
+                                 "start": str(i["start"]),
+                                 "stop": str(i["stop"]),
+                                 "gene": str(i["gene"]),
+                                 "transcript": str(i["transcript"]),
+                                 "exon": str(i["exon"])})
+    return outfile
+
+
+#Function to be run from other scripts
```
Comment (hannasoderstrom1) [inline workflows/scripts/insilico_coverage/anybelow_sample.py:78]: could be nice to have a better description of what the function does

File: workflows/scripts/insilico_coverage/anybelow_sample.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    args = parser.parse_args()
+
+#    logging.getLogger().setLevel(logging.INFO)
+
+    outfiles = [args.csv_output_10x, args.csv_output_20x]
+    limits = [10, 20]
+
+    for limit, outfile in zip(limits, outfiles):
+        fileloc = write_merged(merge_adjacent(read_input(args.csv_input, limit)), outfile)
+        print(f"The file for {limit}X is located at {fileloc.name}")
```
Comment (hannasoderstrom1) [inline workflows/scripts/insilico_coverage/anybelow_sample.py:109]: this is the same code as in function alt_main? why not just use the function?

File: workflows/scripts/insilico_coverage/stats_for_pdf.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("my_input", nargs="?", type=str, help="Full path to input")
+    args = parser.parse_args()
+    logging.getLogger().setLevel(logging.INFO)
+
+    
+
+if __name__ == "__main__":
+    main()
```
Comment (hannasoderstrom1) [inline workflows/scripts/insilico_coverage/stats_for_pdf.py:28]: this script doesn't do anything? i guess it's not done?

**Review by hannasoderstrom1 — APPROVED** [review]
would be nice to have more descriptions of what the functions do

**Comment by alvaralmstedt** [comment]
Too old

**Pull request closed without merging**